apps/vmq_server/src/vmq_server_cli.erl

```diff
@@ -143,32 +143,53 @@ vmq_server_metrics_cmd() ->
+                TextLines =
                 lists:foldl(
                   fun({#metric_def{type = Type,
                                    labels = _Labels,
                                    name = Metric,
                                    description = Description}, Val}, Acc) ->
                           SType = atom_to_list(Type),
                           SMetric = atom_to_list(Metric),
-                          SVal =
+                          Lines =
                               case Val of
-                                  V when is_integer(V) ->
-                                      integer_to_list(V);
-                                  V when is_float(V) ->
-                                      float_to_list(V)
+                                  V when is_number(V) ->
+                                      [clique_status:text(
+                                        lists:flatten([SType, ".", SMetric, " = ", number_to_list(V)]))];
+                                  {Count, Sum, Buckets} when is_map(Buckets) ->
+                                      HistogramLines =
+                                      [
+                                       clique_status:text(
+                                         lists:flatten([SType, ".", SMetric, "_count = ", number_to_list(Count)])),
+                                       clique_status:text(
+                                         lists:flatten([SType, ".", SMetric, "_sum = ", number_to_list(Sum)]))
+                                      ],
+
+                                      maps:fold(
+                                        fun
+                                            (Bucket, BucketCnt, HistAcc) ->
+                                                [clique_status:text(
+                                                   lists:flatten([SType, ".", SMetric, "_bucket_",
+                                                                  case Bucket of
+                                                                      infinity -> "infinity";
+                                                                      _ -> number_to_list(Bucket)
+                                                                  end ," = ", number_to_list(BucketCnt)]))
+                                                 | HistAcc]
+                                        end, HistogramLines, Buckets)
+
                               end,
-                          Line = [SType, ".", SMetric, " = ", SVal],
                           case Describe of
                               true ->
-                                  [clique_status:text(lists:flatten(["# ", Description])),
-                                   clique_status:text(lists:flatten([Line, "\n"]))|Acc];
+                                  [clique_status:text(lists:flatten(["# ", Description]))
+                                   | [[[Lines | clique_status:text("\n")]] | Acc]];
```

Comment:
Fails on dialyzer here. I created a PR to this branch https://github.com/dergraf/vernemq/pull/4 fixing that and there tried to clean up a bit so we only return iolists in all branches and `clique_status:text/1` and `lists:flatten/1` then only have to be called once.

Note, I haven't tested the output of the histogram data in the cli as I didn't know how to easily get such a metric.

Reply:
The output will look similar to this:

```
histogram.metadata_get_microseconds_bucket_infinity = 176
histogram.metadata_get_microseconds_bucket_1000000 = 176
histogram.metadata_get_microseconds_bucket_100000 = 176
histogram.metadata_get_microseconds_bucket_10000 = 176
histogram.metadata_get_microseconds_bucket_1000 = 175
histogram.metadata_get_microseconds_bucket_100 = 172
histogram.metadata_get_microseconds_bucket_10 = 0
histogram.metadata_get_microseconds_count = 176
histogram.metadata_get_microseconds_sum = 7279
```

Reply:
Tried it already with the swc-leveled branch

Reply:
The implemented histogram is a cumulative histogram, as this simplifies the handling for Prometheus (I guess, because they expect the histogram to be cumulative)